dp-cluster-setup-utility.py

```diff
@@ -1743,24 +1761,55 @@ def __init__(self):
     self.dp_instance = None
 
   def execute(self):
+    
     self.dp_instance = self.get_dp_instance()
     dp = self.dp_instance
     if not dp.version.startswith("1.3"):
       print("Registering CM Based cluster is not supported in DP %s" % dp.version)
       return 1
     print "\nTell me about Cloudera Manager Instance"
     cm = ClouderaManager(user.url_input('CM URL', 'cm.url'), user.credential_input('CM admin', 'cm.admin'))
+
     if not CMPrerequisites(cm).satisfied():
       return 1
     
-    if not dp.check_cm(cm):
-      return 1
+    clusters_resp_from_dp = dp.check_cm(cm)
 
-    print 'Registering cluster to DataPlane...'
-    response = dp.register_cm(cm, user)
-    if not response:
+    if not clusters_resp_from_dp:
       return 1
-    return self.handle_registration_response(response)
+    
+    clusters_registered = [cluster.get("name") for cluster in clusters_resp_from_dp if not cluster.get("isUnregistered")]
+    clusters_not_registered = [cluster.get("name") for cluster in clusters_resp_from_dp if cluster.get("isUnregistered")]
+    clusters_to_register = []
+
+    print "Total clusters managed by Cloudera Manager Instance : %s" % cm.total_clusters
+    if cm.total_clusters == 1:
+      clusters_to_register = clusters_not_registered
+    elif cm.total_clusters > 1:
+      print "Clusters already registered in DataPlane : %s" % ','.join([cluster for cluster in clusters_registered])
+      print "Clusters which can be registered in DataPlane : %s" % ','.join([cluster for cluster in clusters_not_registered])
+      if len(clusters_not_registered) > 0:
+        install_all = user.decision('%s y/n' % "Register all", "cm.register_all", default=False)
+        if install_all:
+          clusters_to_register = clusters_not_registered
+        else:
+          user_provided_clusters = []
+          cluster_input_file = user.input('Enter full path of file containing cluster names ', 'cm.cluster_file')
```

Comment:
We will need some more informative text for this flow. Can improve after we this patch in.